### PR TITLE
Add a launcher option to the tester

### PR DESCRIPTION
**Pull request by marcinz**
The tester needs to know about the `--launcher` command line option for multi-rank testing.

### CODE REVIEW

**Comment by bryevdv** [comment]
@marcinz this looks good, if you are interested there are some pure unit tests for the tester itself that could probably be updated:

https://github.com/nv-legate/legate.core/tree/branch-23.09/tests/unit/legate/tester

These are run with plain `pytest tests/unit/legate/tester`

**Comment by marcinz** [comment]
@bryevdv If the tests look good and you approve, then I'll merge.

File: tests/unit/legate/tester/test_args.py [hunk 1]
```diff
@@ -66,6 +68,18 @@ def test_ompthreads(self) -> None:
     def test_numamem(self) -> None:
         assert m.parser.get_default("numamem") == DEFAULT_NUMAMEM
 
+    def test_ranks(self) -> None:
+        assert m.parser.get_default("ranks") == DEFAULT_RANKS_PER_NODE
```
Comment (bryevdv) [inline tests/unit/legate/tester/test_args.py:72]: FYI @manopapad recently renamed this to `ranks_per_node` in the driver, maybe we should change here to match?

File: legate/tester/stages/_linux/gpu.py [hunk 2]
```diff
@@ -67,6 +67,15 @@ def shard_args(self, shard: Shard, config: Config) -> ArgList:
...
             ]
+        if config.nodes > 1:
+            args += [
+                "--nodes",
+                str(config.nodes),
+            ]
+        if config.launcher != "none":
+            args += ["--launcher", str(config.launcher)]
+        for extra in config.launcher_extra:
+            args += ["--launcher-extra=" + str(extra)]
```
Comment (bryevdv) [inline legate/tester/stages/_linux/gpu.py:78]: This same block is repeated a few times, maybe factor it to a helper on the base, then
```
args += self._handle_multi_node_args(config)
```
or similar?

**Review by bryevdv — APPROVED** [review]

**Review by bryevdv — APPROVED** [review]

**Pull request merged**